ui/appui-react/src/test/ui-items-provider/UiItemsManager.test.tsx

```diff
@@ -770,6 +773,345 @@ describe("UiItemsManager", () => {
+          )
+        ).lengthOf(0);
+      });
+    });
+
+    describe("status bar items", () => {
+      const provider = {
+        id: "provider1",
+        getStatusBarItems: () => {
+          return [
+            StatusBarItemUtilities.createActionItem(
+              "item1",
+              StatusBarSection.Left,
+              0,
+              "",
+              "",
+              () => undefined
+            ),
+          ];
+        },
+      } satisfies UiItemsProvider;
+
+      it("should provide", () => {
+        UiItemsManager.register(provider);
+
+        expect(
+          UiItemsManager.getStatusBarItems("stage1", StageUsage.General)
+        ).lengthOf(1);
+      });
+
+      it("should provide only for specified stageIds", () => {
+        UiItemsManager.register({
+          ...provider,
+          stageIds: ["stage2"],
+        });
+
+        expect(
+          UiItemsManager.getStatusBarItems("stage1", StageUsage.General)
+        ).lengthOf(0);
+        expect(
+          UiItemsManager.getStatusBarItems("stage2", StageUsage.General)
+        ).lengthOf(1);
+      });
+
+      it("should provide only for specified stageUsages", () => {
+        UiItemsManager.register({
+          ...provider,
+          stageUsages: ["custom"],
+        });
+
+        expect(UiItemsManager.getStatusBarItems("stage1", "custom")).lengthOf(
+          1
+        );
+        expect(
+          UiItemsManager.getStatusBarItems("stage1", StageUsage.General)
+        ).lengthOf(0);
+      });
+    });
+
+    describe("backstage items", () => {
+      const provider = {
+        id: "provider1",
+        getBackstageItems: () => {
+          return [
+            BackstageItemUtilities.createActionItem(
+              "item1",
+              0,
+              0,
+              () => undefined,
+              ""
+            ),
+          ];
+        },
+      } satisfies UiItemsProvider;
+
+      it("should provide", () => {
+        UiItemsManager.register(provider);
+
+        expect(UiItemsManager.getBackstageItems()).lengthOf(1);
+      });
+
+      it.skip("should provide only for specified stageIds", () => {
+        UiItemsManager.register({
+          id: "provider1",
+          stageIds: ["stage2"],
+          getBackstageItems: () => {
+            return [
+              BackstageItemUtilities.createActionItem(
+                "",
+                0,
+                0,
+                () => undefined,
+                ""
+              ),
+            ];
+          },
+        });
+
+        // TODO: stageId of a provider is ignored in this case.
+        const items = UiItemsManager.getBackstageItems();
+        expect(items).lengthOf(0);
+      });
+
+      it.skip("should provide only for specified stageUsages", () => {
+        UiItemsManager.register({
+          ...provider,
+          stageUsages: ["custom"],
+        });
+
+        // TODO:
+        expect(UiItemsManager.getBackstageItems()).lengthOf(0);
```

Comment:
Should we add optional `stageId` and `stageUsage` arguments to `UiItemsManager.getBackstageItems`? Or should we continue to treat backstage items as global (which IMHO complicates API unnecessarily).

Reply:
If that changes anything, I guess we can add them, since they can be empty anyway.

Reply:
Added.